### DISCUSSION TITLE
How to run +publishSigned with different Scala version of modules depending on each other

### DISCUSSION THREAD

**Discussion by tgodzik** [sbt/sbt#6975] [Q&A] [answered]
Some details can be seen in:

https://github.com/scalameta/scalameta/runs/7357091390?check_suite_focus=true

We use sbt-ci-release, which underneath uses `+publishSigned`

Now if you run it then when compiling for 2.12.10 it will complain about not finding the module that is compiled with 2.12.15.

It seems previously, it would just publish with 2.12.10 and get on with it, but probably it's not the correct behaviour, so it's cool that it was shown.

Is there a way to force the task for 2.12.15 be before 2.12.10. And overall is there a way to fix this globally?

**Comment by eed3si9n** [comment, marked as answer]
This is interesting. Thanks for bringing this to our attention.

For the record (logs may get wiped out) the relevant log entry looks like this:

```
[info] Wrote /home/runner/work/scalameta/scalameta/semanticdb/scalac/plugin/target/scala-2.12/semanticdb-scalac_2.12.10-4.5.9+33-a78362e4-SNAPSHOT.pom
[warn] 
[warn] 	Note: Unresolved dependencies path:
[error] sbt.librarymanagement.ResolveException: Error downloading org.scalameta:scalameta_2.12:4.5.9+33-a78362e4-SNAPSHOT
[error]   Not found
[error]   Not found
[error]   not found: /home/runner/.ivy2/localorg.scalameta/scalameta_2.12/4.5.9+33-a78362e4-SNAPSHOT/ivys/ivy.xml
[error]   not found: https://repo1.maven.org/maven2/org/scalameta/scalameta_2.12/4.5.9+33-a78362e4-SNAPSHOT/scalameta_2.12-4.5.9+33-a78362e4-SNAPSHOT.pom
[error]   not found: https://scala-ci.typesafe.com/artifactory/scala-integration/org/scalameta/scalameta_2.12/4.5.9+33-a78362e4-SNAPSHOT/scalameta_2.12-4.5.9+33-a78362e4-SNAPSHOT.pom
....
[error] (semanticdbScalacCore / update) sbt.librarymanagement.ResolveException: Error downloading org.scalameta:scalameta_2.12:4.5.9+33-a78362e4-SNAPSHOT
```

In other words, semanticdbScalacCore being a Scala compiler plugin must be published at full cross `semanticdb-scalac-core_2.12.10`, but it depends on `.dependsOn(scalameta.jvm)`

https://github.com/scalameta/scalameta/blob/a78362e4f143cf39cd05220eeaab28f086fe3147/build.sbt#L110

and I guess `scalameta.jvm` is published with normal binary cross `_2.12`.

Here's something that could work?
1. Make full crossScalaVersions available in all subprojects, including `scalameta`
2. Instead of calling `+publishSigned` make your own command that splits publishing of full-cross subprojects, and binary-cross subprojects.

Here's my take on it - https://github.com/scalameta/scalameta/pull/2784.

**Reply by tgodzik** [reply]
Thanks! I will take a look at it!
